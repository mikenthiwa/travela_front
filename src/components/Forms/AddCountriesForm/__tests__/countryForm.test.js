import React from 'react';
import AddCountryForm from '../AddCountriesForm';
import AddCountryFields from '../FormFieldSets/CountriesDetails';

describe('< AddCountryForm/>', () => {
  let wrapper, onSubmit;
  onSubmit = jest.fn();

  const props = {
    regionId: '1001',
    handleAddCountries: jest.fn(() => {}),
    isLoading: false,
    addingCountries: false,
    countries: [],
    errors: '',
    meta: { pageCount: 1, currentPage: 1 },
    addItem: jest.fn(() => {}),
    removeItem: jest.fn(() => {}),
    fetchCountries: jest.fn(() => {}),
    values: {
      countries: [],
      item: ''
    },
    closeModal: jest.fn()
  };

  beforeEach(() => {
    wrapper = mount(<AddCountryForm {...props} />);
    jest.resetAllMocks();
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a fieldset', () => {
    const fieldsets = wrapper.find('fieldset');
    expect(fieldsets).toHaveLength(2);
  });

  it('renders the country details fieldset', () => {
    const countryDetails = wrapper.find('fieldset.personal-details');
    expect(countryDetails).toHaveLength(1);
  });

  it('should clear the form when the component unmounts', () => {
    const componentWillUnmount = jest.spyOn(
      wrapper.instance(),
      'componentWillUnmount'
    );
    wrapper.unmount();
    expect(componentWillUnmount).toHaveBeenCalledTimes(1);
  });

  it('should add countries to region', () => {
    const wrapper = mount(<AddCountryForm {...props} />);
    const spy = jest.spyOn(wrapper.instance(), 'handleSubmit');
    wrapper
      .find('input')
      .at(0)
      .simulate('change', { target: { value: 'Nigeria' } });
    wrapper.find('.add_button').simulate('click');
    wrapper.find('form').simulate('submit');
    expect(spy).toHaveBeenCalled();
    expect(props.handleAddCountries).toHaveBeenCalledTimes(1);
    expect(props.handleAddCountries).toHaveBeenCalledWith(
      '1001',
      {'countries': ['Nigeria']}
    );
    jest.resetAllMocks();
    wrapper.unmount();
  });

  it('should  call addItem function', () => {
    const wrapper = mount(<AddCountryForm {...props} />);
    const spy = jest.spyOn(wrapper.instance(), 'addItem');
    wrapper
      .find('input')
      .at(0)
      .simulate('change', { target: { value: 'Nigeria' } });
    wrapper.find('.add_button').simulate('click');
    expect(spy).toHaveBeenCalled();
    jest.resetAllMocks();
    wrapper.unmount();
  });

  it('should  call removeItem function', () => {
    const wrapper = mount(<AddCountryForm {...props} />);
    const spy = jest.spyOn(wrapper.instance(), 'removeItem');
    wrapper
      .find('input')
      .at(0)
      .simulate('change', { target: { value: 'Nigeria' } });
    wrapper.find('.add_button').simulate('click');
    wrapper.find('#remove').simulate('click');
    expect(spy).toHaveBeenCalled();
    jest.resetAllMocks();
    wrapper.unmount();
  });

  it('should  call handleCancel function', () => {
    const wrapper = mount(<AddCountryForm {...props} />);
    const spy = jest.spyOn(wrapper.instance(), 'handleCancel');
    wrapper
      .find('input')
      .at(0)
      .simulate('change', { target: { value: 'Nigeria' } });
    wrapper.find('.add_button').simulate('click');
    wrapper.find('#cancel').simulate('click');
    expect(spy).toHaveBeenCalled();
    jest.resetAllMocks();
    wrapper.unmount();
  });

  describe('<AddCountryFields />', () => {
    const props = {
      values: {
        item: '',
        countries: []
      },
      addItem: jest.fn(),
      removeItem: jest.fn(),
      allCountries: [
        'Kenya',
        'Uganda',
        'Tanzania',
        'Nigeria',
        'Rwanda',
        'Congo'
      ]
    };

    beforeEach(() => {
      wrapper = shallow(<AddCountryFields {...props} />);
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
