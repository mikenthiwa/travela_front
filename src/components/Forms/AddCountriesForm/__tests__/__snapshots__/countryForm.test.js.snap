// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< AddCountryForm/> <AddCountryFields /> renders correctly 1`] = `
<fieldset
  className="personal-details"
>
  <div
    className="size_in countries_size_in"
  >
    <div
      className="add_dept"
    >
      <Input
        autoComplete="off"
        choices={
          Array [
            "Kenya",
            "Uganda",
            "Tanzania",
            "Nigeria",
            "Rwanda",
            "Congo",
          ]
        }
        className="department_dropdown"
        label="Country"
        labelNote=""
        name="item"
        onChange={null}
        size=""
        type="filter-dropdown-select"
      />
    </div>
    <button
      className="disable_button"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      Add
    </button>
  </div>
  <div
    className=""
  >
    <table
      className="mdl-js-data-table table__requests"
      style={
        Object {
          "backgroundColor": "white",
        }
      }
    />
  </div>
</fieldset>
`;

exports[`< AddCountryForm/> renders correctly 1`] = `
<form
  className="new-request"
  key=".0"
  onSubmit={[Function]}
>
  <fieldset
    className="personal-details"
  >
    <div
      className="size_in countries_size_in"
    >
      <div
        className="add_dept"
      >
        <div
          className="form-input department_dropdown "
        >
          <label
            htmlFor="item"
          >
            Country
            <span
              style={
                Object {
                  "color": "red",
                }
              }
            >
               * 
            </span>
          </label>
          <div
            style={
              Object {
                "position": "relative",
              }
            }
          >
            <div
              className="value"
              style={
                Object {
                  "width": "",
                }
              }
            >
              <input
                autoComplete="off"
                choices={
                  Array [
                    "Aruba",
                    "Afghanistan",
                    "Angola",
                    "Anguilla",
                    "Åland Islands",
                    "Albania",
                    "Andorra",
                    "United Arab Emirates",
                    "Argentina",
                    "Armenia",
                    "American Samoa",
                    "Antarctica",
                    "French Southern and Antarctic Lands",
                    "Antigua and Barbuda",
                    "Australia",
                    "Austria",
                    "Azerbaijan",
                    "Burundi",
                    "Belgium",
                    "Benin",
                    "Burkina Faso",
                    "Bangladesh",
                    "Bulgaria",
                    "Bahrain",
                    "Bahamas",
                    "Bosnia and Herzegovina",
                    "Saint Barthélemy",
                    "Saint Helena, Ascension and Tristan da Cunha",
                    "Belarus",
                    "Belize",
                    "Bermuda",
                    "Bolivia",
                    "Caribbean Netherlands",
                    "Brazil",
                    "Barbados",
                    "Brunei",
                    "Bhutan",
                    "Bouvet Island",
                    "Botswana",
                    "Central African Republic",
                    "Canada",
                    "Cocos (Keeling) Islands",
                    "Switzerland",
                    "Chile",
                    "China",
                    "Ivory Coast",
                    "Cameroon",
                    "DR Congo",
                    "Republic of the Congo",
                    "Cook Islands",
                    "Colombia",
                    "Comoros",
                    "Cape Verde",
                    "Costa Rica",
                    "Cuba",
                    "Curaçao",
                    "Christmas Island",
                    "Cayman Islands",
                    "Cyprus",
                    "Czechia",
                    "Germany",
                    "Djibouti",
                    "Dominica",
                    "Denmark",
                    "Dominican Republic",
                    "Algeria",
                    "Ecuador",
                    "Egypt",
                    "Eritrea",
                    "Western Sahara",
                    "Spain",
                    "Estonia",
                    "Ethiopia",
                    "Finland",
                    "Fiji",
                    "Falkland Islands",
                    "France",
                    "Faroe Islands",
                    "Micronesia",
                    "Gabon",
                    "United Kingdom",
                    "Georgia",
                    "Guernsey",
                    "Ghana",
                    "Gibraltar",
                    "Guinea",
                    "Guadeloupe",
                    "Gambia",
                    "Guinea-Bissau",
                    "Equatorial Guinea",
                    "Greece",
                    "Grenada",
                    "Greenland",
                    "Guatemala",
                    "French Guiana",
                    "Guam",
                    "Guyana",
                    "Hong Kong",
                    "Heard Island and McDonald Islands",
                    "Honduras",
                    "Croatia",
                    "Haiti",
                    "Hungary",
                    "Indonesia",
                    "Isle of Man",
                    "India",
                    "British Indian Ocean Territory",
                    "Ireland",
                    "Iran",
                    "Iraq",
                    "Iceland",
                    "Israel",
                    "Italy",
                    "Jamaica",
                    "Jersey",
                    "Jordan",
                    "Japan",
                    "Kazakhstan",
                    "Kenya",
                    "Kyrgyzstan",
                    "Cambodia",
                    "Kiribati",
                    "Saint Kitts and Nevis",
                    "South Korea",
                    "Kosovo",
                    "Kuwait",
                    "Laos",
                    "Lebanon",
                    "Liberia",
                    "Libya",
                    "Saint Lucia",
                    "Liechtenstein",
                    "Sri Lanka",
                    "Lesotho",
                    "Lithuania",
                    "Luxembourg",
                    "Latvia",
                    "Macau",
                    "Saint Martin",
                    "Morocco",
                    "Monaco",
                    "Moldova",
                    "Madagascar",
                    "Maldives",
                    "Mexico",
                    "Marshall Islands",
                    "Macedonia",
                    "Mali",
                    "Malta",
                    "Myanmar",
                    "Montenegro",
                    "Mongolia",
                    "Northern Mariana Islands",
                    "Mozambique",
                    "Mauritania",
                    "Montserrat",
                    "Martinique",
                    "Mauritius",
                    "Malawi",
                    "Malaysia",
                    "Mayotte",
                    "Namibia",
                    "New Caledonia",
                    "Niger",
                    "Norfolk Island",
                    "Nigeria",
                    "Nicaragua",
                    "Niue",
                    "Netherlands",
                    "Norway",
                    "Nepal",
                    "Nauru",
                    "New Zealand",
                    "Oman",
                    "Pakistan",
                    "Panama",
                    "Pitcairn Islands",
                    "Peru",
                    "Philippines",
                    "Palau",
                    "Papua New Guinea",
                    "Poland",
                    "Puerto Rico",
                    "North Korea",
                    "Portugal",
                    "Paraguay",
                    "Palestine",
                    "French Polynesia",
                    "Qatar",
                    "Réunion",
                    "Romania",
                    "Russia",
                    "Rwanda",
                    "Saudi Arabia",
                    "Sudan",
                    "Senegal",
                    "Singapore",
                    "South Georgia",
                    "Svalbard and Jan Mayen",
                    "Solomon Islands",
                    "Sierra Leone",
                    "El Salvador",
                    "San Marino",
                    "Somalia",
                    "Saint Pierre and Miquelon",
                    "Serbia",
                    "South Sudan",
                    "São Tomé and Príncipe",
                    "Suriname",
                    "Slovakia",
                    "Slovenia",
                    "Sweden",
                    "Eswatini",
                    "Sint Maarten",
                    "Seychelles",
                    "Syria",
                    "Turks and Caicos Islands",
                    "Chad",
                    "Togo",
                    "Thailand",
                    "Tajikistan",
                    "Tokelau",
                    "Turkmenistan",
                    "Timor-Leste",
                    "Tonga",
                    "Trinidad and Tobago",
                    "Tunisia",
                    "Turkey",
                    "Tuvalu",
                    "Taiwan",
                    "Tanzania",
                    "Uganda",
                    "Ukraine",
                    "United States Minor Outlying Islands",
                    "Uruguay",
                    "United States",
                    "Uzbekistan",
                    "Vatican City",
                    "Saint Vincent and the Grenadines",
                    "Venezuela",
                    "British Virgin Islands",
                    "United States Virgin Islands",
                    "Vietnam",
                    "Vanuatu",
                    "Wallis and Futuna",
                    "Samoa",
                    "Yemen",
                    "South Africa",
                    "Zambia",
                    "Zimbabwe",
                  ]
                }
                className="occupationInput"
                label="Country"
                name="role"
                onChange={[Function]}
                size=""
                type="text"
                value=""
              />
            </div>
          </div>
          <span
            className="error"
          />
        </div>
      </div>
      <button
        className="disable_button"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        Add
      </button>
    </div>
    <div
      className=""
    >
      <table
        className="mdl-js-data-table table__requests"
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </div>
  </fieldset>
  <fieldset
    className={null}
  >
    <div
      className="submit-area"
    >
      <button
        className="bg-btn bg-btn--inactive"
        id="cancel"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
      <button
        className="bg-btn bg-btn--active"
        disabled={true}
        id="submit"
        onClick={[Function]}
        type="submit"
      >
        <undefined />
      </button>
    </div>
  </fieldset>
</form>
`;
